container/container.go

```diff
@@ -105,26 +106,65 @@ func (c *Container) FindContainer(namespace []string) (types.ContainerJSON, erro
 	return c.client.ContainerInspect(ctx, containerID)
 }
 
-// Status returns the status of a docker container and servcie.
+// Status returns the status of the container based on the docker container and docker service
+// If any error happen the status will be `UNKNOWN`
+// Otherwise the following status will be applied:
+//   - RUNNING: When the container is running in docker (whatever the status of the service)
+//   - STARTING: When the service is running but the container is not yet started
+//   - STOPPED: When neither the container nor the service are running in docker
 func (c *Container) Status(namespace []string) (StatusType, error) {
-	_, err := c.FindService(namespace)
-	if docker.IsErrNotFound(err) {
-		return STOPPED, nil
+	container, err := c.containerExists(namespace)
+	if err != nil {
+		return UNKNOWN, err
 	}
+	service, err := c.serviceExists(namespace)
 	if err != nil {
 		return UNKNOWN, err
 	}
 
-	container, err := c.FindContainer(namespace)
-	if docker.IsErrNotFound(err) {
-		return STARTING, nil
+	statuses := []struct {
+		container bool
+		service   bool
+		status    StatusType
+	}{
+		{service: true, container: true, status: RUNNING},
+		{service: true, container: false, status: STARTING},
+		{service: false, container: true, status: RUNNING}, // This is actually stopping
+		{service: false, container: false, status: STOPPED},
 	}
-	if err != nil {
-		return UNKNOWN, err
+
+	for _, s := range statuses {
+		if s.container == container && s.service == service {
+			return s.status, nil
+		}
 	}
+	return UNKNOWN, nil // This should never be reached but it's better than a panic :)
+}
+
+// containerExists return true if the docker service can be found, return false otherwise and return
+// any errors that are not `NotFound` errors
+func (c *Container) containerExists(namespace []string) (bool, error) {
+	container, err := c.FindContainer(namespace)
+	if err == nil && container.State != nil &&
+		xstrings.SliceContains([]string{"exited", "dead"}, container.State.Status) {
+		return false, nil
+	}
+	return presenceHandling(err)
+}
+
+// serviceExists return true if the docker container can be found, return false otherwise and return
+// any errors that are not `NotFound` errors
+func (c *Container) serviceExists(namespace []string) (bool, error) {
```

Comment:
// serviceExists checks if corresponding container for service namespace can be found.